### PR TITLE
fix!: Simplify emitted declarations for plugins

### PR DESCRIPTION
**Pull request by Fryuni**

### CODE REVIEW

File: .changeset/witty-poems-pull.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+"astro-integration-kit": minor
+---
+
+Simplify emitted declarations for plugins.
```
Comment (florian-lefebvre) [inline .changeset/witty-poems-pull.md:5]:
```suggestion
Simplifies emitted declarations for plugins.
```

**Review by florian-lefebvre — CHANGES_REQUESTED** [review]
Can you add the breaking change to the docs?

File: .changeset/witty-poems-pull.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+```ts
+declare const hookProviderPlugin: astro_integration_kit.Plugin<"hook-provider", {
+    'astro:config:setup': {
+        doThing: (thing: string) => void;
+    };
+};
+```
+
+As shown above, this simplification also removes the unneded re-reclaration of the plugin hook input, which is a breaking change if you are declaring your generic explicitly.
```
Comment (BryceRussell) [inline .changeset/witty-poems-pull.md:41]:
```suggestion
As shown above, this simplification also removes the unneeded re-declaration of the plugin hook input, which is a breaking change if you are declaring your generic explicitly.
```

**Review by BryceRussell — CHANGES_REQUESTED** [review]

**Review by florian-lefebvre — APPROVED** [review]

File: docs/src/content/docs/getting-started/upgrade-guide.mdx [hunk 3]
```diff
@@ -7,6 +7,30 @@ import { Tabs, TabItem } from '@astrojs/starlight/components';
...
+## `0.13.0`
+
+---
+
+### `Plugin` type simplified
+
+The new Plugin signature added back on [`0.9.0`](#new-plugin-signature-and-defineplugin) returned the need for emitted declarations to include the full type for hook parameters from Astro (which was removed on [`0.7.0`](#plugins-types)).
+This was added along with support for a plugin to define new utilities for multiple hooks at once.
+
+Now, the Plugin generics have been simplified again to allow such support without requiring to replicate Astro's hook parameters types. Just like in `0.7.0`, this should be non-breaking for any plugin relying on type inference but plugins with explicitly declared signatures need to update the following:
```
Comment (BryceRussell) [inline docs/src/content/docs/getting-started/upgrade-guide.mdx:19]:
```suggestion
The new Plugin signature added back in [`0.9.0`](#new-plugin-signature-and-defineplugin) returned the need for emitted declarations to include the full type for hook parameters from Astro (which was removed on [`0.7.0`](#plugins-types)).
This was added along with support for a plugin to define new utilities for multiple hooks at once.

Now, the Plugin generics have been simplified again to allow for such support without having to replicate Astro's hook parameters types. Just like in `0.7.0`, this should be non-breaking for any plugin relying on type inference but plugins with explicitly declared signatures need to update the following:
```

**Review by BryceRussell — APPROVED** [review]
LGTM!

**Pull request merged**